xdsl/backend/riscv/register_allocation.py

```diff
@@ -99,6 +126,14 @@ def _free(self, reg: SSAValue) -> None:
                 available_regs.append(reg_name)
 
     def allocate_registers(self, module: ModuleOp) -> None:
+        if self.exclude_preallocated:
```

Comment:
so this will remove the preallocated registers from the set of all available registers, as opposed to just for the values that are live at the same time? Seems a bit coarse as a mechanism, or at least coarser than I expected. Maybe I'm missing the point of this functionality, but it feels like it would be good to add a big doc comment to `self.exclude_preallocated` to explain the objective.